### PR TITLE
Feature/1314 add data element geographic coordinates

### PR DESCRIPTION
**Pull request by KHORCASH**

### CODE REVIEW

**Comment by jacobkellas** [comment]
@KHORCASH let's see if we can wire this up to use the phone's GPS.  I believe you had a branch that did something similar to this a year ago or so.

**Comment by jacobkellas** [comment]
AB#2379

**Comment by KHORCASH** [comment]
See: https://github.com/SanDiegoCountySheriff/Cal-RIPA/pull/653/commits/ca9aa5c905e51401c2471432396f411974ca98ee

Latest commit adds:
- RipaLocation async function to check for geo availability score, disables 'Current Location' button if availability score is > 200 (brought over from branch 1570)
- moved Lat and Long input text fields out of Other location option section and is now under block# and street
- Added index.js validation of coordinates from getCurrentLocation function to match RipaLocation lat and long validation rules. (allow max of 3 numbers after the decimal point)
- Added unit test for Geo availability function

File: UI/src/components/molecules/RipaLocation.vue [hunk 1]
```diff
@@ -288,9 +336,16 @@ export default {
       },
     },
 
-    isGeolocationAvailable() {
-      return navigator.geolocation
-    },
+    // isGeolocationAvailable() {
```
Comment (jacobkellas) [inline UI/src/components/molecules/RipaLocation.vue:339]: Remove commented code

File: UI/src/components/molecules/RipaLocation.vue [hunk 2]
```diff
@@ -444,6 +538,9 @@ export default {
         value: textValue.replace('.', ''),
       })
     }
+
+    this.geolocationAvailable = await this.isGeolocationAvailable()
+    console.log('geoavail:', this.geolocationAvailable)
```
Comment (jacobkellas) [inline UI/src/components/molecules/RipaLocation.vue:543]: Remove console.log

File: UI/src/components/molecules/RipaLocation.vue [hunk 3]
```diff
@@ -517,6 +614,17 @@ export default {
     handleSaveFavorite() {
       this.$emit('on-save-location-favorite', this.model.location)
     },
+
+    isGeolocationAvailable() {
+      return new Promise(resolve => {
+        navigator.geolocation.getCurrentPosition(position => {
+          const accuracyScore = position.coords.accuracy
+          console.log(accuracyScore)
```
Comment (jacobkellas) [inline UI/src/components/molecules/RipaLocation.vue:622]: Remove console.log

File: UI/src/store/index.js [hunk 4]
```diff
@@ -622,13 +622,46 @@ export default new Vuex.Store({
...
+                  )
+                }
+
+                if (position.coords.longitude !== null) {
+                  longDecimal = String(position.coords.longitude).slice(
+                    String(position.coords.longitude).lastIndexOf('.') + 1,
+                  )
+                }
+
+                console.log(latDecimal)
```
Comment (jacobkellas) [inline UI/src/store/index.js:645]: Remove console.log

**Review by jacobkellas — CHANGES_REQUESTED** [review]
Haven't pulled the branch yet to test, but just a few cosmetic changes.

File: UI/src/components/molecules/RipaLocation.vue [hunk 5]
```diff
@@ -335,6 +390,42 @@ export default {
       ]
     },
 
+    latitudeRules() {
```
Comment (jacobkellas) [inline UI/src/components/molecules/RipaLocation.vue:393]: Lat and Long rules don't seem to be working.  I'm getting the error message after clicking "current location"

File: UI/src/components/molecules/RipaLocation.vue [hunk 6]
```diff
@@ -133,6 +143,38 @@
         </v-col>
       </v-row>
 
+      <template v-if="this.model.stopVersion === 2">
+        <ripa-subheader text="-- or --"></ripa-subheader>
+
+        <v-row no-gutters>
```
Comment (jacobkellas) [inline UI/src/components/molecules/RipaLocation.vue:149]: Let's discuss if this should be at the top.  Also when I click current location, the "more location options" switch turns off and hides the field.

**Review by jacobkellas — CHANGES_REQUESTED** [review]

**Review by jacobkellas — APPROVED** [review]

**Comment by jacobkellas** [comment]
@KHORCASH actually this isn't quite done, I forgot about a couple bugs we found last week.  Let me see if I can compile a list and we'll go over this today.

**Review by jacobkellas — CHANGES_REQUESTED** [review]
Bug fixes

**Review by jacobkellas — APPROVED** [review]

**Pull request merged**